wgpu-native/src/device.rs

```diff
@@ -500,19 +510,21 @@ pub extern "C" fn wgpu_device_create_buffer_mapped(
     desc: &resource::BufferDescriptor,
     mapped_ptr_out: *mut *mut u8
 ) -> BufferId {
-    let buffer = device_create_buffer(device_id, desc);
+    let mut desc = desc.clone();
+    desc.usage |= resource::BufferUsageFlags::MAP_WRITE;
+    let mut buffer = device_create_buffer(device_id, &desc);
 
     let device_guard = HUB.devices.read();
     let device = &device_guard[device_id];
 
-    if let Ok(ptr) = unsafe { device.raw.map_memory(&buffer.memory, 0..(desc.size as u64)) } {
+    let read = desc.usage.contains(resource::BufferUsageFlags::MAP_READ);
```

Comment:
there is no point in creating a buffer that is instantly mapped for reading in WebGPU: we don't support persistent mapping, so the only thing that would be readable from it is zeroes